tesseract/tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_utils.h

```diff
@@ -148,36 +148,40 @@ struct KDLChainData
   std::vector<std::string> active_link_list; /**< List of link names that move with changes in joint values */
   Eigen::MatrixX2d joint_limits;             /**< Joint limits */
   std::map<std::string, int> segment_index;  /**< A map from chain link name to kdl chain segment number */
+  std::vector<std::pair<std::string, std::string>> chains; /**< The chains used to create the object */
 };
 
 /**
  * @brief Parse KDL chain data from the scene graph
  * @param results KDL Chain data
  * @param scene_graph The Scene Graph
- * @param base_name The base link name of chain
- * @param tip_name The tip link name of chain
+ * @param chains A vector of link pairs that represent a chain which all get concatenated together
  * @return True if successful otherwise false
  */
 inline bool parseSceneGraph(KDLChainData& results,
                             const tesseract_scene_graph::SceneGraph& scene_graph,
-                            const std::string& base_name,
-                            const std::string& tip_name)
+                            const std::vector<std::pair<std::string, std::string>>& chains)
 {
-  results.base_name = base_name;
-  results.tip_name = tip_name;
-
   if (!tesseract_scene_graph::parseSceneGraph(scene_graph, results.kdl_tree))
   {
     CONSOLE_BRIDGE_logError("Failed to parse KDL tree from Scene Graph");
     return false;
   }
 
-  if (!results.kdl_tree.getChain(results.base_name, results.tip_name, results.robot_chain))
+  results.chains = chains;
+  results.base_name = chains.front().first;
```

Comment:
Should the docs mention that the first link in the chain is the base and the last is the tip?

Practically, this means that the order you list them in the SRDF matters, correct?

Reply:
Updated.